### ISSUE TITLE
GetDataTable Nullable Convertible Properties 

### ISSUE THREAD

**Issue by NoCodeMonkey** [borisdj/EFCore.BulkExtensions#187]
Hi,

You check for null types on property.PropertyType in the DataTable GetDataTable method but you do not check it if the property belongs to the tableInfo.ConvertibleProperties. Would it be possible to support for nullable/DBNull there also, otherwise we get the error DataSet does not support System.Nullable<>.

Thanks

**Comment by borisdj** [comment]
Could you elaborate more and give an example or Test where it can be reproduced.

**Comment by vslee** [comment]
Here is an example:
`System.NotSupportedException: 'DataSet does not support System.Nullable<>.'`

```c#
using EFCore.BulkExtensions;
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace BulkCopyNullableConversionTest226
{
	class Program
	{
		static async Task Main(string[] args)
		{
			using (var myContext = new MyContext())
			{
				int numberOfEntities = 10;
				var entities = new List<Sample>();
				for (int i = 1; i <= numberOfEntities; i++)
				{
					entities.Add(new Sample() { });
				}
				await myContext.BulkInsertAsync(entities);
			}
		}
	}

	public class MyContext : DbContext
	{
		public DbSet<Sample> Samples { get; set; }
		protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
		{
			optionsBuilder.UseSqlServer(@"Data Source=" + System.Environment.MachineName +
				";Initial Catalog=bulkCopyNullableConversionTest;" + "Integrated Security=True" +
				";Connect Timeout=15;Encrypt=False;TrustServerCertificate=True;ApplicationIntent=ReadWrite;MultiSubnetFailover=False");
			optionsBuilder.EnableSensitiveDataLogging();
			optionsBuilder.ConfigureWarnings(w => w.Log(
				Microsoft.EntityFrameworkCore.Diagnostics.CoreEventId.SensitiveDataLoggingEnabledWarning));
		}
		protected override void OnModelCreating(ModelBuilder modelBuilder)
		{
			modelBuilder
				.Entity<Sample>()
				.Property(e => e.DT)
				.HasConversion(v => v.HasValue ? v.Value.ToUniversalTime() : v,
					v => v.HasValue ? DateTime.SpecifyKind(v.Value, DateTimeKind.Utc) : v);
		}
	}

	public class Sample
	{
		public int Id { get; set; }
		public DateTime? DT { get; set; }
	}
}
```

**Comment by borisdj** [comment]
Fixed with 2.5.1
